### PR TITLE
Opt in plugins easily

### PR DESCRIPTION
**Pull request by naggingant**
Been thinking how it is possible to easily opt in and out plugins.
This way could possibly improve the first part.

Any suggestion is much appreciated!
@PuKoren @robjweiss

### CODE REVIEW

**Comment by robjweiss** [comment]
Right now, you select the plugins you want to use in the constructor, is that right?

File: index.js [hunk 1]
```diff
@@ -1,31 +1,39 @@
...
+    reconnection: plugins.Reconnection,
+    duplex: plugins.Duplex,
+    encoding: plugins.Encoding,
+    rpc: plugins.RPC,
+    confirm: plugins.Confirm,
+    retry: plugins.Retry
+};
+
+const resolvePlugins = (plugins) => plugins
+    .map((plugin) => typeof plugin === 'string' ? new Plugins[plugin]() : plugin);
```
Comment (PuKoren) [inline index.js:16]: Maybe you can assert that the plugin exists before trying to instantiate it, so the error would be clear for the lib user

**Comment by naggingant** [comment]
@robjweiss yes that is correct.
This change is intended to also accept strings, so it instantiates the represented plugins with default options.

File: index.js (hunk 1 above)
Comment (naggingant) [reply index.js:16]: Yes it should be there, I believe.
Thanks.

**Comment by naggingant** [comment]
Now either of them is allowed:
```node
const amqp = require('amqp-node');

(async () => {

    // traditional
    const client = await amqp.connect(url, options);

    // recommended
    const client = new amqp.Client(url, options);
    await client.start();

    // immediate start
    const client = await amqp.Client.start(url, options);

})();
```

**Comment by naggingant** [comment]
I need to rework on the whole plugin system anyway, so I'll just merge now.
Let me know if you have any idea or question :hand:

**Pull request merged**